src/ossl/ecc.rs

```diff
@@ -709,3 +714,162 @@ impl Verify for EccOperation {
         Ok(self.output_len)
     }
 }
+
+fn ckd_to_digest_name_slice(mech: CK_EC_KDF_TYPE) -> &'static [u8] {
+    match mech {
+        CKD_SHA1_KDF | CKD_SHA1_KDF_SP800 => OSSL_DIGEST_NAME_SHA1,
+        CKD_SHA224_KDF | CKD_SHA224_KDF_SP800 => OSSL_DIGEST_NAME_SHA2_224,
+        CKD_SHA256_KDF | CKD_SHA256_KDF_SP800 => OSSL_DIGEST_NAME_SHA2_256,
+        CKD_SHA384_KDF | CKD_SHA384_KDF_SP800 => OSSL_DIGEST_NAME_SHA2_384,
+        CKD_SHA512_KDF | CKD_SHA512_KDF_SP800 => OSSL_DIGEST_NAME_SHA2_512,
+        CKD_SHA3_224_KDF | CKD_SHA3_224_KDF_SP800 => OSSL_DIGEST_NAME_SHA3_224,
+        CKD_SHA3_256_KDF | CKD_SHA3_256_KDF_SP800 => OSSL_DIGEST_NAME_SHA3_256,
+        CKD_SHA3_384_KDF | CKD_SHA3_384_KDF_SP800 => OSSL_DIGEST_NAME_SHA3_384,
+        CKD_SHA3_512_KDF | CKD_SHA3_512_KDF_SP800 => OSSL_DIGEST_NAME_SHA3_512,
+        // TODO BLAKE2B digests
+        _ => &[],
+    }
+}
+
+fn ckd_to_digest_name(mech: CK_EC_KDF_TYPE) -> KResult<Vec<u8>> {
+    let digest = ckd_to_digest_name_slice(mech);
+    let mut v = Vec::with_capacity(digest.len());
+    v.extend_from_slice(digest);
+    Ok(v)
+}
+
+pub const X962KDF_NAME: &[u8; 8] = b"X963KDF\0";
+pub const NONE_NAME: &[u8; 1] = b"\0";
+fn ckd_to_type_slice(mech: CK_EC_KDF_TYPE) -> &'static [u8] {
+    match mech {
+        CKD_SHA1_KDF | CKD_SHA224_KDF | CKD_SHA256_KDF | CKD_SHA384_KDF
+        | CKD_SHA512_KDF | CKD_SHA3_224_KDF | CKD_SHA3_256_KDF
+        | CKD_SHA3_384_KDF | CKD_SHA3_512_KDF => X962KDF_NAME,
+        // TODO BLAKE2B digests ?
+        _ => NONE_NAME,
+    }
+}
+
+fn ckd_to_type(mech: CK_EC_KDF_TYPE) -> KResult<Vec<u8>> {
+    let kdf = ckd_to_type_slice(mech);
+    let mut v = Vec::with_capacity(kdf.len());
+    v.extend_from_slice(kdf);
+    Ok(v)
+}
+
+fn new_pkey_ctx_from_obj(pkey: &mut EvpPkey) -> KResult<EvpPkeyCtx> {
+    Ok(EvpPkeyCtx::from_ptr(unsafe {
+        EVP_PKEY_CTX_new_from_pkey(
+            get_libctx(),
+            pkey.as_mut_ptr(),
+            std::ptr::null(),
+        )
+    })?)
+}
+
+impl Derive for ECDHOperation<'_> {
+    fn derive(
+        &mut self,
+        key: &Object,
+        template: &[CK_ATTRIBUTE],
+        _mechanisms: &Mechanisms,
+        objfactories: &ObjectFactories,
+    ) -> KResult<(Object, usize)> {
+        if self.finalized {
+            return err_rv!(CKR_OPERATION_NOT_INITIALIZED);
+        }
+        self.finalized = true;
+
+        let mut params = OsslParam::with_capacity(6).set_zeroize().add_int(
+            name_as_char(OSSL_EXCHANGE_PARAM_EC_ECDH_COFACTOR_MODE),
+            if self.cofactor { 1 } else { -1 } as i32,
+        )?;
+        if self.kdf != CKD_NULL {
+            params = params
+                .add_utf8_string(
+                    name_as_char(OSSL_EXCHANGE_PARAM_KDF_TYPE),
+                    &ckd_to_type(self.kdf)?,
+                )?
+                .add_utf8_string(
+                    name_as_char(OSSL_EXCHANGE_PARAM_KDF_DIGEST),
+                    &ckd_to_digest_name(self.kdf)?,
```

Comment:
As above just use:
```
.add_const_c_string(
    name_as_char(OSSL_EXCHANGE_PARAM_KDF_TYPE),
    name_as_char(ckd_to_digest_name(self.kdf)),
)?
```

where ckd_to_digest_name just returns the const buffer name from ossl/bindings like `mech_type_to_digest_name()` (in ossl/common.c) does.